src/index.ts

```diff
@@ -103,6 +103,18 @@ export class PublicKey {
    * Pseudo-homomorphic Paillier multiplication
    */
   multiply(c: BigInteger.BigNumber, k: BigInteger.BigNumber) {
+    // Insecure naive multiplication: c^0 % n^2 === 1
+    // Ensures we return an encrypted 0
+    if (k.toString() === '0') {
+      return this.encrypt(toBigInteger(0));
+    }
+    // // Insecure naive multiplication: c^1 % n^2 === c
+    // // Ensures we return a different encrypted c
+    if (k.toString() === '1') {
+      const encryptedZero = this.encrypt(toBigInteger(0));
+      return this.addition(toBigInteger(c), encryptedZero);
+    }
+    
     // c is ciphertext. k is either a cleartext message (number) or a scalar
     return toBigInteger(c).modPow(k, this.n2);
```

Comment:
Doesn't `.modPow(0, x)` return 1 and `y.modPow(1, x)` return `y % x`? If so, is this any different than doing `return this.addition(toBigInteger(c).modPow(k, this.n2), BigInteger.zero)` for the entire function?

Reply:
What I'm trying to do here is ensure the `multiply` returns a new ciphertext (when multiplied by `0`) or a refreshed ciphertext (re-encryption when multiplied by `1`)

As it stands, when multiplying by a `0` the returned result is a `0n` instead of a ciphertext representing `0n`. Similarly, when multiplying c by `1`, the function returns the same exact ciphertext (same underlying BitInt) that was passed in instead of a refreshed cipher.

If there's another way to do this using `modPow`, I'd be happy to see that implementation.

Reply:
The ciphertext after multiplying naively by zero is actually `1n` because anything raised to the power of 0 is 1.

Reply:
I've updated this PR to reflect @willclarktech comment.